### PR TITLE
Update API doc

### PR DESCRIPTION
**Pull request by wy65701436**
1, Add post user API.
2, Add change user password API.
3, Update post project return code to 201.

### CODE REVIEW

File: docs/swagger.yaml [hunk 1]
```diff
@@ -438,6 +484,47 @@ paths:
...
         500:
           description: Unexpected internal errors.
+  /users/{user_id}/password:
+    put:
+      summary: Change the password on a user that already exists.
+      description: |
+        This endpoint is for user to update password. Users with the admin 
+		role can change any user's password. Guest users can change only 
+		their own password.
+      parameters:
```
Comment (wknet123) [inline docs/swagger.yaml]: There can not be multi-line of parameters that against the Swagger yaml specification.
You may refer this [http://swagger.io/specification/#parameterObject](http://swagger.io/specification/#parameterObject)

Comment (wy65701436) [inline docs/swagger.yaml]: It could be caused by copy/paste. Already remove the multi-line parameters. Thanks.

**Comment by wknet123** [comment]
It seems the Swagger YAML file you edit still not pass by checking with the validator of Swagger Parser, you may need to correct them. You can check it with [http://editor.swagger.io/#/](http://editor.swagger.io/#/)

**Comment by wy65701436** [comment]
@wknet123  can you help on the travis CI error? I didn't update any code except the swagger file.

../../docker/distribution/registry/api/errcode/register.go:74: undefined: http.StatusTooManyRequests

**Comment by wknet123** [comment]
It may caused by the update of distribution codes, we'll make a workaround.

**Comment by wy65701436** [comment]
Thanks @wknet123 , you can just let me know when it done. Then I can re-commit the change.

**Comment by wknet123** [comment]
We've bumped up the version of Golang base image,  it should be fine now.

**Comment by wknet123** [comment]
LGTM

**Pull request merged**